scripts/validate_documentation.py

```diff
@@ -0,0 +1,227 @@
+        dict
+            A dictionary containing all the information obtained from
+            validating the page.
+
+        Notes
+        -----
+        The errors codes are defined as:
+        - First two characters: Type of errors:
+           * DT: Error with unwanted node constellations the doctree
+           * WS: Issues regarding whitespace characters
+        - Last two characters: Numeric error code
+        """
+
+    with open(DOCTREE_PATH.format(page), 'r+b') as file:
+        doctree = pickle.load(file)
+
+    with open(RST_PATH.format(page), 'r') as file:
+        raw_doc = file.read()
+
+    checker = DocumentChecker(raw_doc, doctree)
+    return checker.validate()
+
+
+def validate_all(exclude_patterns):
+    """
+    Execute the validation of all pages, and return a dict with the
+    results.
+
+    Parameters
+    ----------
+    exclude_patterns : List[str] or None
+        If provided, the pages that match with one of these patterns
+        will be ignored. If None, all pages will be validated.
+
+    Returns
+    -------
+    dict
+        A dictionary with an item for every page containing
+        all the validation information.
+    """
+
+    result = {}
+    for root, dirs, files in os.walk(DOCUMENTATION_SOURCE):
+        _, base_dir = root.split('../doc/source')
+        for file in files:
+            docname, ext = os.path.splitext(file)
+            if not ext == '.rst':
+                continue
+            page = os.path.join(base_dir, docname)
+            for pattern in exclude_patterns:
+                if fnmatch(page, pattern):
+                    continue
+
+            result[page] = validate_one(page)
+    return result
+
+
+def main(page, errors, output_format, exclude_patterns=None):
+    if page:
+        reports = {page: validate_one(page)}
+    else:
+        reports = validate_all(exclude_patterns=exclude_patterns)
+    return report(reports, output_format=output_format, errors=errors)
+
+
+if __name__ == '__main__':
+    format_opts = 'default', 'json', 'azure'
+    func_help = ('document to validate (e.g. io) '
+                 'if not provided, all documents are validated')
+    argparser = argparse.ArgumentParser(
+        description='validate pandas documentation')
+    add = argparser.add_argument
+    add('document',
+        nargs='?',
+        default=None,
+        help=func_help)
+    add('--format', default='default', choices=format_opts,
+        help='format of the output when validating '
+             'multiple documents (ignored when validating one).'
+             'It can be {}'.format(str(format_opts)[1:-1]))
+    add('--errors', default=None,
+        help='comma separated '
+             'list of error codes to validate. By default it '
+             'validates all errors (ignored when validating '
+             'a single document)')
+    add('--exclude', default=None,
+        help='comma separated '
+             'patterns of pages to exclude. By default it '
```

Comment:
From the help message I don't understand how I should use this. `--exclude=io.rst,api.rst`?